src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -169,22 +170,25 @@ public function set($id, $service)
         }
 
         if (isset($this->aliases[$id])) {
-            unset($this->aliases[$id]);
+            @trigger_error(sprintf('Replacing the "%s" (alias of "%s") service is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0.', $id, $this->aliases[$id]), E_USER_DEPRECATED);
+        } elseif (isset($this->services[$id])) {
```

Comment:
I think `isset` is not enough to check if a previous service exists (see `Container::has()`).
But I don't know if we should authorize replacing a non instantiated service or not.